modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AMQPConstructStore.java

```diff
@@ -126,7 +126,13 @@ public void addQueue(AndesQueue queue, boolean isLocal) throws AndesException {
      * @throws AndesException
      */
     public void removeQueue(String queueName, boolean isLocal) throws AndesException {
-        if (isLocal) {
+
+        // If the queue deletion call was originally received by this node, the queue should be removed from the
+        // database as well. Else, only the entry of the queue mapping cache should be removed.
+        if (!isLocal){
+            messageStore.removeCachedQueue(queueName);
```

Comment:
I don't this we have to pass "isLocal" and at runtime decide on the pathe we should take. We know if isLocal true or false at the compile time. Therefore we should have used two methods for this. WDYT?